core/Tracker/Model.php

```diff
@@ -357,23 +357,26 @@ public function findVisitor($idSite, $configId, $idVisitor, $fieldsToRead, $shou
         // 		this page view to the wrong visitor, but this is better than creating artificial visits.
         // 2) there is a visitor ID and we trust it (config setting trust_visitors_cookies, OR it was set using &cid= in tracking API),
         //      and in these cases, we force to look up this visitor id
-        $whereCommon = "visit_last_action_time >= ? AND visit_last_action_time <= ? AND idsite = ?";
-        $bindSql = array(
+        $configIdWhere = "visit_last_action_time >= ? AND visit_last_action_time <= ? AND idsite = ?";
+        $configIdbindSql = array(
```

Comment:
One of the reason we added the 30min lookback window WHERE expression was to reduce performance problems: IIRC large scale Matomo instances have performance problems with selecting visitors in the past without WHERE visit_last_action_time limitations... (even though in theory the index should make it fast, I don't know why it wasn't the case before.)  some of these SELECT queries were slow and ended up piling up and tracking didn't work anymore. 

-> Maybe this problem would now be fixed (eg. a Mysql performance improvement etc.), but we need need to test the new code on a very large instance (under high traffic load, eg. 100 tracking api requests per second, without QueuedTracking) to check whether it works fast.

(If we merged this without testing performance it would possibly result in tracker performance regressions which is risky and high impact on users)

Reply:
@tsteur ran the queries on a large instance and seemed to be ok. Will let him follow up.

Reply:
MySQL shouldn't even use that index re time and I would suspect it makes it rather slower but might be wrong. 

We were wondering in chat whether there are any privacy problems with this eg around 13 months etc.

Reply:
As far as I see there is no privacy implications. (as long as there is a matching Visitor ID in the past, then the visitor ID will be the same in both visits and this PR actually makes data more accurate)

Todo:

* we just need to update one sentence in this FAQ that mentions the look back window https://matomo.org/faq/general/faq_21418/
* and also the comment of INI setting `window_look_back_for_visitor` 

to explain that for "Visitor ID" matching we look back in the whole history while for the Fingerprint we only look back window_look_back_for_visitor ie. 30min

Now I see that there is likely no performance regression here so this change should be fine :+1:

Reply:
Made the change to INI config. Suggested faq change:

```
In all cases, when “looking for visits” in the past, we look back window_look_back_for_visitor seconds. By default, Matomo (Piwik) will look back 30 minutes (default value for window_look_back_for_visitor INI setting). This setting can be changed: learn more.
```

to

```
When matching visits by visitor/user ID, Matomo looks through the entire history of visits that are currently in the database. When matching visits by device fingerprint (config_id), we only look back window_look_back_for_visitor seconds. By default, this value is set to 30 minutes. This setting can be changed: learn more.
```

Reply:
Thanks. i've already updated FAQ so we don't forget it later: https://matomo.org/faq/general/faq_21418/